lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -49,7 +52,26 @@ export class NativeInteractionClient extends BaseInteractionClient {
         const reqTimestamp = TimeUtils.nowSeconds();
         const response: object = await this.nativeMessageHandler.sendMessage(messageBody);
         const validatedResponse: NativeResponse = this.validateNativeResponse(response);
-        return this.handleNativeResponse(validatedResponse, nativeRequest, reqTimestamp);
+
+        // generate the authentication result
+        let authResult: AuthenticationResult;
+        try {
+            authResult = await this.handleNativeResponse(validatedResponse, nativeRequest, reqTimestamp);
+            // end the perf measurement for success
+            nativeATMeasurement.endMeasurement({
+                success: true,
+                isNativeBroker: true
+            });
+        } catch (e) {
+            // end the perf measurement for failure
+            nativeATMeasurement.endMeasurement({
+                success: false,
+                isNativeBroker: true
+            });
+            throw e;
+        }
+
+        return authResult;
```

Comment:
Nitpick: it can be a bit cleaner to put this onto the promise chain returned, e.g.: 

```suggestion
        return this.handleNativeResponse(validatedResponse, nativeRequest, reqTimestamp)
          .then((result: AuthenticationResult) => {
            nativeATMeasurement.endMeasurement({
                success: true,
                isNativeBroker: true
            });
            return result;
          })
          .catch((error) => {
            nativeATMeasurement.endMeasurement({
                success: false,
                isNativeBroker: true
            });
            throw e;
          });
```